go/services/reports/internal/brokers/report_request_failed.go

```diff
@@ -0,0 +1,86 @@
+package brokers
+
+import (
+	"os"
+	"time"
+
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/envs"
+	"github.com/Magpie-Monitor/magpie-monitor/pkg/message-broker"
+	"go.uber.org/fx"
+	"go.uber.org/zap"
+)
+
+var REPORT_REQUEST_FAILED_TOPIC_KEY = "REPORT_REQUEST_FAILED_BROKER_TOPIC"
+
+type ReportRequestError = string
+
+const (
+	ReportRequestError_Validation ReportRequestError = "VALIDATION_ERROR"
+	ReportRequestError_Timeout    ReportRequestError = "TIMEOUT"
+	ReportRequestError_Internal   ReportRequestError = "INTERNAL_ERROR"
+)
+
+type ReportRequestFailed struct {
+	CorrelationId string             `json:"correlationId"`
+	ErrorType     ReportRequestError `json:"reportRequest"`
```

Comment:
Can enum type with funciton that'd return corresponding string value be used here instead?

Reply:
It might, but I don't see a reason to do this in this case. This struct is pretty much a DTO which is directly encoded and pushed to a error queue. If I understand you correctly, this would mean that consumer would recieve a error code, instead of error string description. 
Do you mind explaining you reasoning a bit more, in case I misunderstood your intentions?

Reply:
My proposal was to use enum with String() method that'd map error to it's string value that'd be returned to client, instead of having it defined in consts. If you feel that its not needed then feel free to merge :D